### ISSUE TITLE
How to use with can-value?

### ISSUE THREAD

**Issue by matthewp** [canjs/can-stache-element#29]
With can-component you can bind to observations when instantiating new instances like so:

```js
new MyComponent({
  viewModel: {
     age: value.from(this, 'age')
  }
});
```

In StacheDefineElement I tried to set a property like this:

```js
let el = new MyElement();
el.age = value.from(this, 'age');
```

However this threw because it was setting the prop to a wrong type (observation). To fix this I had to change the property to `type.Any`.

I thought maybe I'd need to use [can-bind](https://canjs.com/doc/can-bind.html) directly but that would be tricky to do in my case, it's inside a getter so there isn't a place for teardown.

This seems like something that StacheDefineElement should have a solution for.

**Comment by justinbmeyer** [comment]
```


let el = new MyElement();

var bind = new Bind({
    child: value.to(el,"age"),
    parent: value.from(this, 'age')
}).start();

el.disconnectedCallback = function(){
    bind.teardown();
}

domMutate.onNodeRemoved(el, );


el.age = 

var vm = new ViewModel();

for(var prop in  props) }{
    vm[prop]= props[prop]
}


bindings(MyElement,{
    toChild: value.from(this,"prop")
    toParent: value.to(this,"prop")
    twoWay: value.bind (this,"prop"),
    value: 123
})


new Element().bindings({})

bindings = function(ElementConstructor, componentInitVM) {


    var el =  new ElementConstructor();
    el.connectedCallback = function(){
        canReflect.eachKey(componentInitVM, function(parent, propName) {

                    var canGetParentValue = parent != null && !!parent[getValueSymbol];
                    var canSetParentValue = parent != null && !!parent[setValueSymbol];

                    // If we can get or set the value, then we’ll create a binding
                    if (canGetParentValue === true || canSetParentValue) {

                        // Create an observable for reading/writing the viewModel
                        // even though it doesn't exist yet.
                        var child = stacheBindings.getObservableFrom.viewModel({
                            name: propName,
                        }, bindingContext, bindingSettings);

                        // Create the binding similar to what’s in can-stache-bindings
                        var canBinding = new Bind({
                            child: child,
                            parent: parent,
                            queue: "domUI",

                            //!steal-remove-start
                            // For debugging: the names that will be assigned to the updateChild
                            // and updateParent functions within can-bind
                            updateChildName: "update viewModel." + propName + " of <" + el.nodeName.toLowerCase() + ">",
                            updateParentName: "update " + canReflect.getName(parent) + " of <" + el.nodeName.toLowerCase() + ">"
                            //!steal-remove-end
                        });

                        bindings.push({
                            binding: canBinding,
                            siblingBindingData: {
                                parent: {
                                    source: "scope",
                                    exports: canGetParentValue
                                },
                                child: {
                                    source: "viewModel",
                                    exports: canSetParentValue,
                                    name: propName
                                }
                            }
                        });

                    } else {
                        // Can’t get or set the value, so assume it’s not an observable
                        initialVMData[propName] = parent;
                    }
                });

                // Initialize the viewModel.  Make sure you
                // save it so the observables can access it.
                var initializeData = stacheBindings.behaviors.initializeViewModel(bindings, initialVMData, function(properties){
                    el = new ElementConstructor();
                    el.render(properties);
                    return el;
                }, bindingContext);
                teardown = function() {
        			for (var attrName in initializeData.onTeardowns) {
        				initializeData.onTeardowns[attrName]();
        			}
        		};

                return el;
        }
        el.disconnectedCallback = fumnction(){
            el.innerHTML = "";
            el._rendered = false;
            teardown();
        }
}
```

**Comment by matthewp** [comment]
Resolved by #31
